engine/src/main/java/org/camunda/bpm/engine/impl/management/SetJobRetriesBuilderImpl.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.camunda.bpm.engine.impl.management;
+
+import static org.camunda.bpm.engine.impl.util.EnsureUtil.ensureNotEmpty;
+import static org.camunda.bpm.engine.impl.util.EnsureUtil.ensureNotNull;
+
+import java.util.Date;
+import java.util.List;
+
+import org.camunda.bpm.engine.ProcessEngineException;
+import org.camunda.bpm.engine.impl.ProcessEngineLogger;
+import org.camunda.bpm.engine.impl.cmd.CommandLogger;
+import org.camunda.bpm.engine.impl.cmd.SetJobRetriesCmd;
+import org.camunda.bpm.engine.impl.interceptor.CommandExecutor;
+import org.camunda.bpm.engine.management.SetJobRetriesBuilder;
+
+public class SetJobRetriesBuilderImpl implements SetJobRetriesBuilder {
+
+  protected final static CommandLogger LOG = ProcessEngineLogger.CMD_LOGGER;
+
+  protected final CommandExecutor commandExecutor;
+
+  protected String jobId;
+  protected List<String> jobIds;
+  protected String jobDefinitionId;
+  protected Integer retries;
+  protected Date dueDate;
+
+  boolean byJobId = false;
+  boolean byJobIds = false;
+  boolean byJobDefinitionId = false;
+
+  public SetJobRetriesBuilderImpl(CommandExecutor commandExecutor, int retries) {
+    this.commandExecutor = commandExecutor;
+    this.retries = retries;
+  }
+
+  @Override
+  public SetJobRetriesBuilder jobId(String jobId) {
+    this.jobId = jobId;
+    byJobId = true;
+    return this;
+  }
+
+  @Override
+  public SetJobRetriesBuilder jobIds(List<String> jobIds) {
+    this.jobIds = jobIds;
+    byJobIds = true;
+    return this;
+  }
```

Comment:
I believe that this would complicate things. For example, when I receive a list of job ids from an API call I have to convert them into an array before using the builder. I think the consistent pattern in our API is to use `List<String>` for collections of ids.

Reply:
When I search through our code base for `String...` I see 148 places in non-impl classes where we use this pattern. In the first step, we could only add `String... jobIds` since this is less code to write and test on our side, and in the future, when we receive feedback that this API is annoying, we could still add a `List<String> jobIds` method.

Reply:
I know that we use the varargs pattern in our API. My point is that using it here feels like we are mixing `String[]` and `List<String>`. Our query APIs return lists and we have an [API method to set job retries by lists of job ids](https://github.com/camunda/camunda-bpm-platform/blob/8c273b76b032682a07d0cf824e451b881d06c0a4/engine/src/main/java/org/camunda/bpm/engine/ManagementService.java#L823-L840).
It would require converting from list to array in some places and we would need to include null checks there as well which are currently done at a central point. 

It feels like more overhead for us now and later for the users.